can/can_filter_mode.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef HAL_EXTENSION_CAN_FILTER_MODE_HPP
+#define HAL_EXTENSION_CAN_FILTER_MODE_HPP
+
+#include "main.h"
+
+namespace halex {
+
+enum class CAN_FilterMode: uint8_t {
+    PATH_ONE_TYPE_STD_OR_EXT_ID_GROUP = 0x10,
+    PATH_TWO_TYPE_STD_OR_EXT_ID       = 0x12,
+    PATH_TWO_TYPE_STD_ID_GROUP        = 0x13,
+    PATH_FOUR_TYPE_STD_ID             = 0x14,
+    PATH_ALL_ID                       = 0x15
+};
```

Comment:
```suggestion
enum class CAN_FilterMode: uint8_t {
    ONE_TYPE_STD_OR_EXT_ID_GROUP = 0x00,
    TWO_TYPE_STD_OR_EXT_ID       = 0x01,
    TWO_TYPE_STD_ID_GROUP        = 0x02,
    FOUR_TYPE_STD_ID             = 0x03,
    ALL_ID                       = 0x04
};
```

[WANT] 可能なのであれば、０からの連番にしたいです。また、それぞれどういうモードなのか簡単にコメントで書いてくれると嬉しいです。PATH は共通なので消して、意味的に必要なのであれば FilterMode を FilterPathMode などにしたいです

Reply:
eff9030a55fec676e81459f1cbb692c7a30d4d8e で削除